src/Universe.cpp

```diff
@@ -535,27 +534,45 @@ Cell* Universe::findCell(LocalCoords* coords) {
       coords->setCell(cell);
 
       /* MATERIAL type Cell - lowest level, terminate search for Cell */
-      if (cell->getType() == MATERIAL) {
-        coords->setCell(cell);
-        return_cell = cell;
-        return return_cell;
-      }
+      if (cell->getType() == MATERIAL)
+        return cell;
 
       /* FILL type Cell - Cell contains a Universe at a lower level
        * Update coords to next level and continue search */
       else if (cell->getType() == FILL) {
 
-        LocalCoords* next_coords;
-
-        if (coords->getNext() == NULL)
-          next_coords = new LocalCoords(coords->getX(), coords->getY(),
-                                        coords->getZ());
-        else
-          next_coords = coords->getNext();
+        LocalCoords* next_coords =
+            new LocalCoords(coords->getX(), coords->getY(), coords->getZ());
+        next_coords->setPhi(coords->getPhi());
+
+        /* Apply translation to position in the next coords */
+	if (cell->isTranslated()){
+	  double* translation = cell->getTranslation();
+	  double new_x = coords->getX() + translation[0];
+	  double new_y = coords->getY() + translation[1];
+	  double new_z = coords->getZ() + translation[2];
+	  next_coords->setX(new_x);
+	  next_coords->setY(new_y);
+	  next_coords->setZ(new_z);
+	}
+
+        /* Apply rotation to position and direction in the next coords */
+	if (cell->isRotated()){
+	  double x = coords->getX();
+	  double y = coords->getY();
+	  double z = coords->getZ();
+	  double* matrix = cell->getRotationMatrix();
+	  double new_x = matrix[0] * x + matrix[1] * y + matrix[2] * z;
+	  double new_y = matrix[3] * x + matrix[4] * y + matrix[5] * z;
+	  double new_z = matrix[6] * x + matrix[7] * y + matrix[8] * z;
+	  next_coords->setX(new_x);
+	  next_coords->setY(new_y);
+	  next_coords->setZ(new_z);
+          next_coords->incrementPhi(cell->getPsi() * M_PI / 180.);
```

Comment:
Weird, it doesn't show up that way in emacs. I removed all whitespace and added it back in so perhaps it will show up properly now.